src/core/actions/ActionGenerator.ts

```diff
@@ -5,37 +5,97 @@ export class ActionGenerator {
+
+    /**
+     * Shorthand to generate an action of type 'intent'.
+     *
+     * @param {ActionType} type
+     * @param {string} name
+     * @param {string} origin
+     * @param {ActionPayload} [payload]
+     * @param {Range} [position]
+     * @returns {Intent}
+     */
+    static intent(name: string, origin: string, payload?: ActionPayload, position?: Range): Intent {
+        return this.make('intent', name, origin, payload, position) as Intent;
+    }
+    /**
+     * Shorthand to generate an action of type 'primitive'.
+     *
+     * @param {ActionType} type
+     * @param {string} name
+     * @param {string} origin
+     * @param {ActionPayload} [payload]
+     * @param {Range} [position]
+     * @returns {Primitive}
+     */
+    static primitive(
+        name: string,
+        origin: string,
+        payload?: ActionPayload,
+        position?: Range,
+    ): Primitive {
+        return this.make('primitive', name, origin, payload, position) as Primitive;
     }
-    remove(target?: DOMElement): void {
-        this.dispatcher.dispatch({
-            id: 'primitive.remove', // todo: automize
-            name: 'remove',
-            type: 'primitive',
-            target: target,
-            origin: arguments.callee.caller.name,
-        });
+    /**
+     * Shorthand to generate an action of type 'command'.
+     *
+     * @param {ActionType} type
+     * @param {string} name
+     * @param {string} origin
+     * @param {ActionPayload} [payload]
+     * @param {Range} [position]
+     * @returns {Command}
+     */
+    static command(
+        name: string,
+        origin: string,
+        payload?: ActionPayload,
+        position?: Range,
+    ): Command {
+        return this.make('command', name, origin, payload, position) as Command;
     }
-    update(value: DOMElement, target?: DOMElement): void {
-        this.dispatcher.dispatch({
-            id: 'primitive.update', // todo: automize
-            name: 'update',
-            type: 'primitive',
-            payload: {
-                element: value,
-            },
-            target: target,
-            origin: arguments.callee.caller.name,
-        });
+    /**
+     * Return a properly formatted Action.
+     *
+     * @param {ActionType} type
+     * @param {string} name
+     * @param {string} origin
+     * @param {ActionPayload} [payload]
+     * @param {Range} [position]
+     * @returns {Intent|Primitive|Command}
+     */
+    static make(
+        type: ActionType,
+        name: string,
+        origin: string,
+        payload?: ActionPayload,
+        position?: Range,
+    ): Action {
+        const action: Action = {
+            id: this._makeID(type, name),
+            type: type,
+            name: name,
+            origin: origin,
+        };
+        if (payload) {
+            action.payload = payload;
+        }
+        if (position) {
+            action.position = position;
+        }
+        return action;
+    }
```

Comment:
When I see all these functions, I wonder why we don't simply use a `actionParams` parameter, which follow the interface of these function arguments, and smply add the missing `id` property via `make`. No more `if`s. You want a payload ? You set it. You didn't set a payload ? We won't set a payload. If the payload is an object, you'll have to make the call multiline anyway, and it will look quite strange with every parameter listed in the function call. Plus, you want to set `position` but not `payload` ? You're screwed, you'll have to pass a useless value like null or undefined for `payload`... :(